### DISCUSSION TITLE
Python project for wallet management

### DISCUSSION THREAD

**Discussion by geoffmunn** [classic-terra/core#152] [Show and tell]
Hi everyone,

I've taken the old terra.py project and updated it for Luna Classic purposes. It was actually in a pretty good state as long as you picked the correct release numbers.

I've also created some utility scripts in Python to allow for multi-wallet management. You can:

- make reward withdrawals from validators,
- swap USTC for LUNC, and 
- send anything back to the same validator (redelegate).

This allows you to do most things that the Terra Station app did, but without having to rely on the TFL team or Terra Rebels to keep things updated. Also, there are minor fee differences which I couldn't figure out - it looks like this Python project is slightly cheaper.

If/when we get a DEX set up, I'll add swap support for non-native tokens and coins which should open a world of opportunities :)

Is there a reason why the terra.py wasn't cloned into the classic-terra repo with everything else? I'm happy to maintain it but it would be nice to give it some visiblity here.

Relevant links:
- **Terra.py (modified)**: [https://github.com/geoffmunn/terra.py](https://github.com/geoffmunn/terra.py)
- **Utility scripts**: [https://github.com/geoffmunn/utility-scripts](https://github.com/geoffmunn/utility-scripts)

**Comment by nghuyenthevinh2000** [comment]
thanks you so much for this, let's wait for L1 to review it

**Comment by geoffmunn** [comment]
I have just released version 1.2.0, which is a stability release. It now has a gas discovery function, so all transactions should be guaranteed to work at the minimum possible price.
You can now send LUNC to any address via the send_transactions.py script, and check your balances with get_balances.py.

The next release will provide full delegation support with validators, allowing you to undelegate and switch validators. At this point, you don't need the station app at all.

https://github.com/geoffmunn/utility-scripts/releases/tag/v1.2.0

**Reply by ZaradarBH** [reply]
Lets bring this over to the org and make  you an outside contrib with maintainer access? It would make it a lot easier  for people to find everything in one place :)

**Comment by dellacostamoveis-code** [comment]
Uma idéia para captar mais apoiadores 
Fazer uma parceria com o Pix  Brasileiro oferendo a compra de 1 lunc por cada transação do Pix.
São 150.000.000 de transações por dia um volume de Milhões em casa dia.
Sendo o valor lunc acessível com certeza haverá novos traders.
Começar com o nubank.
